bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -177,15 +182,16 @@ public void initialize() {
         initJob = scheduler.schedule(() -> {
             boolean start = true;
             try {
-                initializeThingConfig();
-                logger.debug("{}: Device config: Device address={}, HTTP user/password={}/{}, update interval={}",
-                        thingName, config.deviceAddress, config.userId.isEmpty() ? "<non>" : config.userId,
-                        config.password.isEmpty() ? "<none>" : "***", config.updateInterval);
-                logger.debug(
-                        "{}: Configured Events: Button: {}, Switch (on/off): {}, Push: {}, Roller: {}, Sensor: {}, CoIoT: {}, Enable AutoCoIoT: {}",
-                        thingName, config.eventsButton, config.eventsSwitch, config.eventsPush, config.eventsRoller,
-                        config.eventsSensorReport, config.eventsCoIoT, bindingConfig.autoCoIoT);
-                start = initializeThing();
+                if (initializeThingConfig()) {
+                    logger.debug("{}: Device config: Device address={}, HTTP user/password={}/{}, update interval={}",
+                            thingName, config.deviceAddress, config.userId.isEmpty() ? "<non>" : config.userId,
+                            config.password.isEmpty() ? "<none>" : "***", config.updateInterval);
+                    logger.debug(
```

Comment:
You could implement an override ToString method in the config class. That is more readable, and can easily be re-used anwhere you need to log the used config.

Reply:
yup, changed